internal/app/go-filecoin/paymentchannel/integration_test.go

```diff
@@ -0,0 +1,126 @@
+	"github.com/filecoin-project/go-filecoin/internal/pkg/message"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/repo"
+	th "github.com/filecoin-project/go-filecoin/internal/pkg/testhelpers"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/types"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/vm/actor"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/vm/state"
+)
+
+func TestConstructInitActor(t *testing.T) {
+	ctx := context.Background()
+	fms := paychtest.NewFakeActorInterface(t, ctx, abi.NewTokenAmount(9999))
+	assert.NotNil(t, fms)
+}
+
+// TestCreatePaymentChannel tests that the RetrievalClient can call through to the InitActor and
+// successfully cause a new payment channel to be created.
+func TestCreatePaymentChannel(t *testing.T) {
+	ctx := context.Background()
+	balance := abi.NewTokenAmount(1000000)
+	chainBuilder, bs, genTs := testSetup2(ctx, t)
+	ds := dss.MutexWrap(datastore.NewMapDatastore())
+	fms := paychtest.NewFakeActorInterface(t, ctx, balance)
+	rt := fms.Runtime
+	root, err := chainBuilder.GetTipSetStateRoot(genTs.Key())
+	require.NoError(t, err)
+
+	channelAmt := abi.NewTokenAmount(101)
+	_, client, miner, paychID, paych := fms.StubCtorSendResponse(channelAmt)
+	fakeProvider := message.NewFakeProvider(t)
+	fakeProvider.Builder = chainBuilder
+	clientActor := actor.NewActor(builtin.AccountActorCodeID, balance, root)
+	fakeProvider.SetHead(genTs.Key())
+	fakeProvider.SetActor(client, clientActor)
+
+	viewer := paychtest.NewFakeStateViewer(t)
+
+	pchMgr := pch.NewManager(context.Background(), ds, fms, fms, viewer)
+
+	viewer.AddActorWithState(paych, client, miner, address.Undef)
+
+	rmc := retrievalmarketconnector.NewRetrievalMarketClientFakeAPI(t)
+
+	rcnc := retrievalmarketconnector.NewRetrievalClientConnector(bs, fakeProvider, rmc, pchMgr)
+	assert.NotNil(t, rcnc)
+	tok, err := encoding.Encode(genTs.Key())
+	require.NoError(t, err)
+
+	res, err := rcnc.GetOrCreatePaymentChannel(ctx, client, miner, channelAmt, tok)
+	require.NoError(t, err)
+	assert.Equal(t, paych, res)
+	var st init_.State
+	rt.GetState(&st)
+	actualIdAddr, err := st.ResolveAddress(adt.AsStore(rt), paych)
+	require.NoError(t, err)
+	require.Equal(t, paychID, actualIdAddr)
+}
+
+func testSetup2(ctx context.Context, t *testing.T) (*chain.Builder, bstore.Blockstore, block.TipSet) {
+	_, builder, genTs, cs, st1 := requireNewEmptyChainStore(ctx, t)
+	rootBlk := builder.AppendBlockOnBlocks()
+	th.RequireNewTipSet(t, rootBlk)
+	require.NoError(t, cs.SetHead(ctx, genTs))
+	root, err := st1.Commit(ctx)
+	require.NoError(t, err)
+
+	// add tipset and state to chainstore
+	require.NoError(t, cs.PutTipSetMetadata(ctx, &chain.TipSetMetadata{
+		TipSet:          genTs,
+		TipSetStateRoot: root,
+		TipSetReceipts:  types.EmptyReceiptsCID,
+	}))
+
+	ds := repo.NewInMemoryRepo().ChainDatastore()
+	bs := bstore.NewBlockstore(ds)
+	return builder, bs, genTs
+}
+
+func TestPaychActorIFace(t *testing.T) {
```

Comment:
This interface will be used for Collect and Settle integration and functional tests